.gitignore

```diff
@@ -1,21 +1,24 @@
-*.swp
-*.pyc
-sphinx/build
-manual/build
+# Hidden files
+.*
+
+# Python byte / compiled / optimized
+*.py[cod]
+__pycache__/
+
+# Build artifacts
+build/
+_build/
+makelog.txt
+
+# Obsolete in-source build artifacts
 manual/pdf
-manual/source/.introduction.rst.swp
 manual/source/nexus.pdf
 manual/source/nxdl_desc.rst
 manual/source/*.table
 manual/source/classes/*/NX*.rst
 manual/source/classes/*/*.nxdl.xml
 manual/source/classes/contributed_definitions/canSAS
-manual/makelog.txt
-/build/
+
+# Unknown
 /python/
-impatient-guide/_build
-.DS_Store
 __github_creds__.txt
-
-# Microsoft VSCode editor configuration
```

Comment:
This is still a good protection to keep.  It prevents unwanted content from creeping in by accident.

Reply:
That's taken care of by the more general

```
# Hidden files
.*
```

Should stop all editors like eclipse, vscode, pycharm, ...